src/commands/dev.ts

```diff
@@ -370,23 +395,36 @@ export async function command(commandOptions: CommandOptions) {
         return;
       }
 
-      if (proxy) {
-        for (const workerConfig of config.scripts) {
-          if (workerConfig.type !== 'proxy') {
-            continue;
+      config.proxy.forEach(([pathPrefix, proxyOptions], i) => {
```

Comment:
This should be moved up outside of the handler that gets called for every request, right? This looks like it would kill performance

Reply:
This was roughly how `http-proxy-middleware` handled it and it just returns if the listener is in place. I will play around with moving it down to the http server listener. I didn't notice any slowdowns in my basic tests

Reply:
I was able to move it down to the HTTP server listener.